### DISCUSSION TITLE
Shell completion

### DISCUSSION THREAD

**Discussion by rgoncalvesse** [zk-org/zk#165] [Help] [answered]
I haven't seen any issues or notices about command completion in zsh/bash. Only completion in the editor via LSP is mentionned.

Am I am missing something or is there still no shell completion?

**Comment by mickael-menu** [comment, marked as answer]
No indeed, this is not supported at the moment.

Could you describe a bit more what stuff you expect should auto complete in the shell?

**Reply by rgoncalvesse** [reply]
Actually, I would separate the completion in two distinct types:

1. Basic completion. Suggesting commands (e.g. new, list, ...) and flags for each of those commands (e.g. --help, -f/--format, ...).
2. Filter completion. Suggesting note title, note id or tags when filtering (e.g. https://github.com/mickael-menu/zk/blob/main/docs/note-filtering.md#filter-by-tags)

For instance, `taskwarrior` suggests a multitude of dates/day/month/events when creating a new task with a due date. For zk, the ultimate completion state could be to suggest similar days/events when filtering notes by creation or modification (see below).

![image](https://user-images.githubusercontent.com/18457507/152557882-4562b5c8-a7ee-4ba6-b97a-55598ea71dc4.png)

**Reply by mickael-menu** [reply]
Yes that would be very useful. For the basic completion I found this lib that could do the trick: https://github.com/WillAbides/kongplete

It might be possible to extend it to support custom completion.
